### ISSUE TITLE
Allow configurable lint check to flag/change commands to uppercase/lowercase or as it currently is mixed-case

### ISSUE THREAD

**Issue by compor** [cheshirekow/cmake_format#239]
Hello,

First, thanks a lot for the work in this project.

## TL;DR
Allow configurable lint check to flag/change commands to uppercase/lowercase or as it currently is mixed-case

## Description

I came across this issue because I recently switch from [`cmake-lint`][1] to, well, `cmakelint` (this project v0.6.13).
I'm using Vim with ALE for linting and will transitioning and trying out this linter I've hooked up both of them in order to get their recommendations. Consider this CMakeLists.txt:

```
cmake_minimum_required(VERSION 3.19)
project(issuereport C CXX)

include(FetchContent)

FetchContent_Declare(
    Catch2
    GIT_REPOSITORY https://github.com/catchorg/Catch2.git
    GIT_TAG v2.13.3)

FetchContent_MakeAvailable(Catch2)

list(APPEND CMAKE_MODULE_PATH ${catch2_SOURCE_DIR}/contrib)
include(Catch)

add_executable(unittests foo.cpp)                                                                                                                                                                                               

target_link_libraries(unittests Catch2::Catch2)

# make tests discoverable via CTest
catch_discover_tests(unittests)
```

I'd like the `FetchContent_MakeAvailable` to stay as or be transformed to `fetchcontent_makeavailable`.
However, `cmake-format` converts it to the former and I haven't been able to find a relevant lint [rule][2] or option in the config to change this.

On the other hand, [`cmake-lint`][1] flags this via its `readability/wonkycase` rule.

Command case-insensitivity is one of the "features" of CMake that I hate and when mixing these it sort of makes it worse for me.
I'm not familiar with a convention, since I've seen projects use all capitals (especially older ones), a mix or all lowercase.
I'm of the latter school and I was wondering if it would be considered if an option for this was made preserving the current default behaviour for consistency with older releases of `cmake_format` & co., but allowing users like me to set this to their preference.

I'd be glad to discuss this further or point me to the right direction if it'd be preferrable to help with this otherwise.

[1]: https://github.com/cmake-lint/cmake-lint
[2]: https://cmake-format.readthedocs.io/en/latest/lint-summary.html

**Comment by cheshirekow** [comment]
I think the option you are looking for is [command_case][1]. The option only applies to cmake-format, there is no corresponding lint rule (yet... as you have noticed in the lint db). I believe that does address your request though (e.g. `cmake-format` will make things consistent). The default is `canonical`. For the case of `FetchContent_MakeAvailable` the wonky case [is canonical][2]. I believe what you want is to change this option to `lower`. 

Canonical spellings for each command are specified by the ["spelling" field][3] for a standard entry. There are also a couple that are implemented as overrides [here][4]. 

[1]: https://cmake-format.readthedocs.io/en/latest/configopts.html#command-case
[2]: https://cmake.org/cmake/help/git-stage/module/FetchContent.html
[3]: https://github.com/cheshirekow/cmake_format/blob/master/cmakelang/parse/funs/standard_modules.py#L83
[4]: https://github.com/cheshirekow/cmake_format/blob/master/cmakelang/parse/funs/standard_funs.py#L10

**Comment by compor** [comment]
Yes, I think it does address my issue partly for now. I somehow hastily and erroneously assumed that the lint and format rule correspondence is there in all the tools.

Let me play a bit around with this and get back to you in order to close this.

How much interest would there be for the corresponding lint rule? Maybe I can muster a PR if I find the time. Any pointers welcome.

Thanks for getting back fast.

**Comment by cheshirekow** [comment]
Adding a command case lint check is reasonable. I think implementation can be fairly straightforward. Let me have a look at adding this in the next batch of changes.
